antarest/study/business/areas/hydro_management.py

```diff
@@ -15,20 +14,27 @@
 
 
 class ManagementOptionsFormFields(FormFieldsBaseModel):
-    inter_daily_breakdown: Optional[Union[StrictFloat, StrictInt]]
-    intra_daily_modulation: Optional[Union[StrictFloat, StrictInt]]
-    inter_monthly_breakdown: Optional[Union[StrictFloat, StrictInt]]
-    reservoir: Optional[StrictBool]
-    reservoir_capacity: Optional[Union[StrictFloat, StrictInt]]
-    follow_load: Optional[StrictBool]
-    use_water: Optional[StrictBool]
-    hard_bounds: Optional[StrictBool]
-    initialize_reservoir_date: Optional[Union[StrictFloat, StrictInt]]
-    use_heuristic: Optional[StrictBool]
-    power_to_level: Optional[StrictBool]
-    leeway_low: Optional[Union[StrictFloat, StrictInt]]
-    leeway_up: Optional[Union[StrictFloat, StrictInt]]
-    pumping_efficiency: Optional[Union[StrictFloat, StrictInt]]
+    inter_daily_breakdown: Optional[float]
+    intra_daily_modulation: Optional[float] = Field(ge=1) # >= 1
+    inter_monthly_breakdown: Optional[float]
+    reservoir: Optional[bool]
+    reservoir_capacity: Optional[float]
+    follow_load: Optional[bool]
+    use_water: Optional[bool]
+    hard_bounds: Optional[bool]
+    initialize_reservoir_date: Optional[int] = Field(ge=0, le=11) # 0 <= x <= 11
+    use_heuristic: Optional[bool]
+    power_to_level: Optional[bool]
+    use_leeway: Optional[bool]
+    leeway_low: Optional[float]
+    leeway_up: Optional[float]
+    pumping_efficiency: Optional[float] = Field(ge=0, le=1) # 0 <= x <= 1
+
+    @validator("*", pre=True)
+    def check_positive(cls, v: Union[float, int]) -> Union[float, int]:
+        if v < 0:
+            raise ValueError("value must be positive")
+        return v
```

Comment:
This validator will be triggered even for boolean fields, which doesn't really make sense. I prefer that we use the `Field` class everywhere, as it will allow us to document the fields in the future using the `description` parameter, especially to indicate the unit used.